conda_build/build.py

```diff
@@ -490,61 +486,55 @@ def is_no_link(no_link, short_path):
 
 
 def get_inode_paths(files, target_short_path, prefix):
-    ensure_list(files)
-    target_short_path_inode = os.stat(join(prefix, target_short_path)).st_ino
+    target_short_path_inode = os.lstat(join(prefix, target_short_path)).st_ino
     hardlinked_files = [sp for sp in files
-                        if os.stat(join(prefix, sp)).st_ino == target_short_path_inode]
+                        if os.lstat(join(prefix, sp)).st_ino == target_short_path_inode]
     return sorted(hardlinked_files)
 
 
-def file_type(path):
-    if isdir(path):
-        return FileType.directory
-    elif islink(path):
-        return FileType.softlink
-    return FileType.hardlink
+def node_type(path):
+    if islink(path):
+        return NodeType.softlink
+    return NodeType.hardlink
 
 
 def build_info_files_json(m, prefix, files, files_with_prefix):
     no_link = m.get_value('build/no_link')
     files_json = []
-    for fi in files:
+    for fi in sorted(files):
         prefix_placeholder, file_mode = has_prefix(fi, files_with_prefix)
         path = os.path.join(prefix, fi)
         file_info = {
             "short_path": get_short_path(m, fi),
             "sha256": sha256_checksum(path),
             "size_in_bytes": os.path.getsize(path),
-            "file_type": getattr(file_type(path), "name"),
+            "node_type": node_type(path).name,
```

Comment:
1) Remove `.name` from the line above.
2) The class below is from `conda._vendor.auxlib.entity`.  Let's put it in `conda_interface.py`.

```
class EntityEncoder(JSONEncoder):
    # json.dumps(obj, cls=SetEncoder)
    def default(self, obj):
       if hasattr(obj, 'dump'):
          return obj.dump()
       elif hasattr(obj, '__json__'):
          return obj.__json__()
       elif hasattr(obj, 'to_json'):
          return obj.to_json()
       elif hasattr(obj, 'as_json'):
          return obj.as_json()
       return JSONEncoder.default(self, obj)
```

3) Add a `__json__` method to `NodeType` that returns `self.name`.  See other enums in `conda/base/constants.py` for examples.